plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -119,6 +119,13 @@ def set_via_env_json
         data.merge!(YAML.safe_load(yaml))
       end
 
+      env = static_env
+      if template.dig(:metadata, :annotations, :"samson/append_tag_to_name") == "true"
+        tag = env.fetch(:TAG)
+        env["TAG_AS_JSON"] = tag.to_json
+        env["NAME_AS_JSON"] = "#{template.dig_fetch(:metadata, :name)}-#{tag}".to_json
```

Comment:
couldn't they configure this in their samson config since it has some interpolation, rather than having to build a whole feature for it?
e.g. set `NAME_AS_JSON=my-project-$TAG` in the config, along with the set-from-env annotations?